icicle/include/icicle/program/program.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+#include "icicle/program/symbol.h"
+
+namespace icicle {
+
+  using InstructionType = uint32_t;
+
+  enum PreDefinedPrograms { AB_MINUS_C, EQ_X_AB_MINUS_C };
```

Comment:
why EQ_X but AB? I mean maybe it's E times Q? the name convention is not really clear. I do understand from the context but still

Reply:
added in comment clarification about those macros